user_tools/src/spark_rapids_tools/tools/top_candidates.py

```diff
@@ -85,12 +86,21 @@ def _generate_output_table_internal(self, output_df: pd.DataFrame) -> pd.DataFra
         """
         Internal implementation to prepare the output table. This can be overridden by the child classes.
         """
+        # Append 'Speedup Category Order' and 'Total Core Seconds' columns to output_df for sorting order
+        # First, sort based on speedup category (decs order). Create and append 'Speedup Category Order' column.
+        speedup_category_order = ['Not Recommended', 'Small', 'Medium', 'Large']
+        output_df['Speedup Category Order'] = \
+            output_df['Estimated GPU Speedup Category'].map({name: i for i, name in enumerate(speedup_category_order)})
+        # Second, within each speedup category, sort based on total core seconds (desc order)
+        output_df = pd.merge(output_df, self.qual_summary[['App ID', 'Total Core Seconds']],
+                             how='left', left_on='App ID', right_on='App ID')
+
+        # Sort columns and select output columns
```

Comment:
I am pretty sure this code is going to change a lot due to extending the heuristics.
Perhaps the sorting logic can be a separate stage?

Reply:
after addressing the review feedbacks, this part of the code looks more concise. If it still looks like sorting logic can be a separate stage, I can change that, thanks.